src/ResourceManager/Sql/Commands.Sql/ManagedInstance/Cmdlet/NewAzureSqlManagedInstance.cs

```diff
@@ -0,0 +1,198 @@
+
+        /// <summary>
+        /// The location in which to create the Managed instance
+        /// </summary>
+        [Parameter(Mandatory = true,
+            HelpMessage = "The location in which to create the Managed instance")]
+        [LocationCompleter("Microsoft.Sql/managedInstances")]
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Managed instance Subnet Id
+        /// </summary>
+        [Parameter(Mandatory = true,
+            HelpMessage = "The Subnet Id to use for Managed instance creation")]
+        [ValidateNotNullOrEmpty]
+        public string SubnetId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Managed instance License Type
+        /// </summary>
+        [Parameter(Mandatory = true,
+            HelpMessage = "Determines which License Type of Sql Azure Managed instance to use")]
+        [ValidateSet(Constants.LicenseTypeBasePrice, Constants.LicenseTypeLicenseIncluded, IgnoreCase = false)]
+        public string LicenseType { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Storage Size in GB for Managed instance
+        /// </summary>
+        [Parameter(Mandatory = true,
+            HelpMessage = "Determines how much Storage size to associate with Managed instance")]
+        [ValidateNotNullOrEmpty]
+        public int StorageSizeInGB { get; set; }
+
+        /// <summary>
+        /// Gets or sets the VCores for Managed instance
+        /// </summary>
+        [Parameter(Mandatory = true,
+            HelpMessage = "Determines how much VCores to associate with Managed instance")]
+        [ValidateNotNullOrEmpty]
+        public int Vcore { get; set; }
+
+        /// <summary>
+        /// Gets or sets the managed instance version
+        /// </summary>
+        [Parameter(Mandatory = true,
+            HelpMessage = "The compute generation for the Sql Azure Managed Instance. e.g. 'GP_Gen4', 'BC_Gen4'.")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet(Constants.GeneralPurposeGen4, Constants.GeneralPurposeGen5, Constants.BusinessCriticalGen4, Constants.BusinessCriticalGen5, IgnoreCase = false)]
+        public string RequestedSkuName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the tags to associate with the Azure Sql Managed Instance
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "The tags to associate with the Azure Sql Managed Instance")]
+        [Alias("Tag")]
+        public Hashtable Tags { get; set; }
+
+        /// <summary>
+        /// The tags to associate with the Azure Sql Managed Instance
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "Do not generate and assign an Azure Active Directory Identity for this Managed instance for use with key management services like Azure KeyVault.")]
+        public SwitchParameter DoNotAssignIdentity { get; set; }
```

Comment:
For SQL Server we have -AssignIdentity, why is it flipped for MI?

Reply:
Same question